src/service-object.ts

```diff
@@ -360,13 +365,28 @@ class ServiceObject<T = any> extends EventEmitter {
    * @param {object} callback.metadata - The metadata for this object.
    * @param {object} callback.apiResponse - The full API response.
    */
-  getMetadata(): Promise<MetadataResponse>;
+  getMetadata(options?: GetMetadataOptions): Promise<MetadataResponse>;
   getMetadata(callback: MetadataCallback): void;
-  getMetadata(callback?: MetadataCallback): Promise<MetadataResponse>|void {
+  getMetadata(options: GetMetadataOptions, callback: MetadataCallback): void;
+  getMetadata(
+      optionsOrCallback?: GetMetadataOptions|MetadataCallback,
+      callback?: MetadataCallback): Promise<MetadataResponse>|void {
+    let options: GetMetadataOptions = {};
+    if (typeof optionsOrCallback === 'function') {
```

Comment:
Consider using [maybeOptionsOrCallback](https://github.com/googleapis/nodejs-common/blob/master/src/util.ts#L756) to simplify this logic!